Install python v2 in PHP images

phil-davis opened the pull request:
PR #138 removed node, npm and yarn. Somehow we lost python v2 by doing that - I guess that it happened to be a dependency of one or more of those removed tools.

But we need python v2 for the caldav/carddav test suites. There is not yet a version of those test suites that passes with python v3.
See https://github.com/apple/ccs-caldavtester/issues/39 and https://github.com/apple/ccs-caldavtester/pull/40 for recent progress on that in the upstream repo.

This PR explicitly installs python v2 in the PHP images, so that we can get oC10 core CI green again.

phil-davis commented:
Note: I think I am right:
- on Ubuntu 18.04 `sudo apt install python` gets python v2
- on Ubuntu 20.04 `sudo apt install python2` gets python v2

xoxys commented:
Uff Python 2 EOL has been announced for so long and there is still stuff was not ported... Not really happy about this change

phil-davis commented:
This PR to the upstream project attempts to move to python v3: https://github.com/apple/ccs-caldavtester/pull/40

I tried running with that, but still got test fails. It needs some work.

I needed to get CI green again in core. "someone" (tm) can decide if we contribute upstream or... to get these test pipelines updated to python v3.

xoxys commented:
But why have we added this test suite to CI now if we exactly know that it is python 2 which is EOL? Or how could have these test passed before we added python2 to this image? Or was the CI red for the last months?

xoxys commented:
Have not seen it before but in general, honestly I need to ask why we choose a GH project that was not updated since 3 years now.

phil-davis commented:
> But why have we added this test suite to CI now if we exactly know that it is python 2 which is EOL? Or how could have these test passed before we added python2 to this image? Or was the CI red for the last months?

This happened today after removing node, npm and yarn from `owncloudci/php`. I guess that python v2 was still a dependency of something in those? And we had been "lucky" that python v2 still happened to be installed. Now we have to explicitly mention it.

xoxys commented:
Ok it gets worse and worse :see_no_evil: Thanks for the backround

phil-davis commented:
> Have not seen it before but in general, honestly I need to ask why we choose a GH project that was not updated since 3 years now.

Because it was chosen 3 or more years ago. "someone" (tm) chose it, and now it is rarely/not maintained!

I raised core issue https://github.com/owncloud/core/issues/39249 and another "someone" can decide what to do next.
